src/WebTelemetryTransport.ts

```diff
@@ -18,6 +18,29 @@ export class WebTelemetryTransportDefault implements WebTelemetryTransport {
     }
 }
 
+export class WebTelemetryTransportWithHeaders implements WebTelemetryTransport {
```

Comment:
Everything is okay, but the naming is too broad. I suggest using `WebTelemetryTransportJson` instead `WebTelemetryTransportWithHeaders`.